invenio_rdm_records/services/permissions.py

```diff
@@ -74,8 +75,8 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
         IfRestricted("files", then_=can_view, else_=can_all),
     ]
     can_get_content_files = [
-        # preview is same as read_files
-        IfFileIsLocal(then_=can_preview, else_=[SystemProcess()])
+        # TODO: is this business logic, rather than permissions?
```

Comment:
It is closer to business logic, but is difficult to delegate to the service/storage... I think @ppanero might remember a bit more on what the discussion was to leave it here.

Reply:
It was highly complex/coupling to do this. We had to do it in the service component and there were use cases that could not be easily abstracted and it was coupling transfer classes, data layer and services. And there was a corner case that was a deadlock (i.e. needed some info to continue but the info needed what was being calculated or it wasn't available at that point in time). However, I do not recall the specifics. #shouldvedocumented